util/tracing/opt_trace.go

```diff
@@ -93,3 +93,34 @@ type CETraceRecord struct {
 	Expr      string
 	RowCount  uint64
 }
+
+// PhysicalOptimizeTraceInfo indicates for the PhysicalOptimize trace information
+type PhysicalOptimizeTraceInfo struct {
+	ID       int                          `json:"id"`
+	TP       string                       `json:"type"`
+	Children []*PhysicalOptimizeTraceInfo `json:"children"`
+	Property string                       `json:"property"`
+
+	// ExplainInfo should be implemented by each implemented LogicalPlan
+	ExplainInfo string `json:"info"`
+
+	BestTask   *TaskInfo   `json:"best"`
+	Candidates []*TaskInfo `json:"candidates"`
+}
+
+type TaskInfo struct {
+	Cost float64 `json:"cost"`
+}
+
+// PhysicalOptimizeTracer indicates the trace for the whole physicalOptimize processing
+type PhysicalOptimizeTracer struct {
+	Root    *PhysicalOptimizeTraceInfo
+	Current *PhysicalOptimizeTraceInfo
+}
```

Comment:
Add a comment that we maintain `PhysicalOptimizeTraceInfo` as a tree and let `Current` become `current`.